Add REST API docs for creating annotations (#2546)
buildkite/docs: pages/apis/rest_api/annotations.md

Required scope: `read_builds`

Success response: `200 OK`

## Create an annotation on a build

Creates an annotation on a build.

>📘
> Note that you need the <a href="/docs/apis/rest-api/builds#build-number-vs-build-id">build number</a> to create annotations, not the build ID.

```bash
curl -X POST "https://api.buildkite.com/v2/organizations/{org.slug}/pipelines/{pipeline.slug}/builds/{build.number}/annotations" \
  -H "Content-Type: application/json" \
  -d '{
    "body": "Hello world!",
    "style": "info",
    "context": "greetings"
  }'
```

```json
{
  "id": "018b8d10-6b5b-4df2-b0ff-dfa2af566050",

>   "context": "greeting",

Reviewer: Nitpick: The request has `greetings` as the `context` but the response drops the `s`. I assume these should match?

```suggestion
  "context": "greetings",
```

^ or the other way around.

  "style": "info",
  "body_html": "<p>Hello world!</p>\n",
  "created_at": "2023-11-01T22:45:45.435Z",
  "updated_at": "2023-11-01T22:45:45.435Z"
}
```

Required [request body properties](/docs/api#request-body-properties):

<table class="responsive-table">
<tbody>
  <tr>
    <th><code>body</code></th>
    <td>
      The annotation's body, as <a href="/docs/agent/v3/cli-annotate#supported-markdown-syntax">HTML or Markdown</a>.
      <p class="Docs__api-param-eg"><em>Example:</em> <code>"My annotation here"</code></p>
    </td>
  </tr>
</tbody>
</table>

Optional [request body properties](/docs/api#request-body-properties):

<table class="responsive-table">
<tbody>
  <tr>
    <th><code>style</code></th>
    <td>

>       The style of the annotation. Can be `success`, `info`, `warning` or `error`

Reviewer: The Markdown formatting isn't picked up in this HTML block:

```suggestion
      The style of the annotation. Can be <code>success</code>, <code>info</code>, <code>warning</code> or <code>error</code>.
```

      <p class="Docs__api-param-eg"><em>Example:</em> <code>"info"</code></p>
    </td>
  </tr>
  <tr>
    <th><code>context</code></th>
    <td>

>       A string value by which to identify the annotation on the build. Only one annotation per build may have any given context value. Useful when appending existing annotations.

Reviewer: I think switching the sentence order helps the explanation:

```suggestion
      A string value by which to identify the annotation on the build. This is useful when appending to an existing annotation. Only one annotation per build may have any given context value.
```

      <p class="Docs__api-param-eg"><em>Example:</em> <code>"coverage"</code></p>
    </td>
  </tr>
  <tr>
    <th><code>append</code></th>
    <td>

>       Whether to append the given body onto the annotation with the same context.

Reviewer: I think using code formatting on `body` will help connect it to the field listed above:

```suggestion
      Whether to append the given <code>body</code> onto the annotation with the same context.
```
